src/hooks/useNavigationFocus.js

```diff
@@ -5,29 +5,27 @@
  */
 
 import { useEffect, useRef } from 'react';
-import { refreshData } from '../pages/dataTableUtilities';
 
 /**
  * Simple custom hook which subscribes to navigation focus event,
- * (navigating to a page), which will refresh data on that page
- * when navigating to it. Main use case is popping a page from the
- * stack and returning to a page whose state has been altered but
- * the UI has not updated/re-rendered.
+ * (navigating to a page), which will invoke the passed callback
+ * when navigating to it.
+ *
+ * Main use case is popping a page from the stack and returning to
+ * a page whose state has been altered but the UI has not updated.
  *
- * - Using this hook will REFRESH ALL DATA on navigating to this page.
- * - ENSURE THE CALLING SCREEN HAS A `refreshData` REDUCER CASE.
  * - Only events for the calling page are subscribed to.
  * - Initial navigation focus event is NOT captured.
  *
- * @param {Func}   dispatch   Page reducer
+ * @param {Func}   callback   callback to invoke on focusing the subscribed screen.
  * @param {Object} navigation react-navigation navigator object.
  */
-export const useNavigationFocusRefresh = (dispatch, navigation) => {
+export const useNavigationFocus = (callback, navigation) => {
   const subscription = useRef(null);
 
   const subscribe = () => {
     if (subscription.current) return;
-    subscription.current = navigation.addListener('willFocus', () => dispatch(refreshData()));
+    subscription.current = navigation.addListener('willFocus', callback);
   };
 
   const unSubscribe = () => {
```

Comment:
Nb: is there a use-case for `subscribe` and `unsubscribe` outside of `useEffect`? If not, should they be left out to indicate they should only be run (automatically) when component is mounted/unmounted and are not intended to be used by user code?

Reply:
I could see the potential for switching between sync modes (i.e. with sensors) or during a process like dispensing where you wouldnt want updates to occur during issuing a prescription but would if you were just on a list, maybe. I don't see the harm in having the option? If one day it is added, there would be potential for regressions in existing code (unlikely, but ya know), but not the other way round?